packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -1195,14 +1196,15 @@ export class AngularCompilerPlugin {
           'AngularCompilerPlugin._emit.ts', diagMode));
 
         if (!hasErrors(allDiagnostics)) {
-          if (this._firstRun || changedTsFiles.size > 20) {
+          if (this._firstRun || changedTsFiles.size > 20 || !this._hadFullJitEmit) {
```

Comment:
I think the correct approach is to never enter the "only 20 files" branch if the last build wasn't successful. We can use `this._emitSkipped` to know this.

Reply:
You are right :)